doc/development/plugins.rst

```diff
@@ -97,169 +92,165 @@ as well as potential further capabilities, by providing the following class attr
-  the resulting expectation value from the device.
+    class CustomDevice(QubitDevice):
+        name = 'My custom device'
+        short_name = 'example.mydevice'
+        pennylane_requires = '0.1.0'
+        version = '0.0.1'
+        author = 'Ada Lovelace'
 
-* :meth:`.Device.var`: This accepts an observable name (as a string), the wires (subsystems)
-  to measure, and the parameters for the observable. It is expected to return the resulting
-  variance of the measured observable value from the device.
+        operations = {"PauliX", "RX", "CNOT"}
+        observables = {"PauliZ", "PauliX", "PauliY"}
 
-* :meth:`.Device.sample`: This accepts an observable name (as a string), the wires (subsystems)
-  to measure, and the parameters for the observable. It is expected to return samples of the
-  measured observable value from the device. Number of samples is determined by attribute
-  :attr:`.Device.shots`.
+            def __init__(self, shots=1024, hardware_options=None):
+                super().__init__(wires=24, shots=shots, analytic=False)
+                self.hardware_options = hardware_options or hardware_defaults
 
-.. note::
+Note that we have also overridden the default shot number.
 
-    Currently, PennyLane only supports measurements that return a scalar value.
+The user can now pass any of these arguments to the PennyLane device loader:
 
-Tensor observables
-~~~~~~~~~~~~~~~~~~
+>>> dev = qml.device("example.mydevice", hardware_options={"t2": 0.1})
+>>> dev.hardware_options
+{"t2": 0.1}
 
-A user may specify multi-wire observables using either :class:`pennylane.Hermitian` and
-providing an Hermitian NumPy array, or by declaring a tensor product of single-wire observables
-using the syntax ``qml.PauliX(0) @ qml.PauliZ(1)``.
 
-If you would like to support tensor observables, make sure to specify ``tensor_observables: True``
-in the :attr:`~.Device._capabilities` dictionary. When a measurement function is called
-with a tensor product of observables, each argument (``observable``, ``wires``, ``par``)
-become lists of equal size.
+Device execution
+----------------
 
-For example, for ``qml.PauliX(0) @ qml.Hermitian(A, [1, 3])``,
-the :meth:`~.Device.expval` method will be called as follows from PennyLane:
+Once all the class attributes are defined, it is necessary to define some required class
+methods, to allow PennyLane to apply operations and measure observables on your device.
 
-.. code-block:: python
+To execute operations on the device, the following methods **must** be defined:
 
-    result = dev.expval(["PauliX, Hermitian"], wires=[[0], [1, 3]], par=[[], [A]])
+.. currentmodule:: pennylane.QubitDevice
 
+.. autosummary::
 
-Advanced execution control
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+    apply
+    probability
 
-Additional flexibility is sometimes required for interfacing with more
-complicated frameworks.
+(Click the name of each method to see more details and examples). The :class:`~.QubitDevice` class
```

Comment:
```suggestion
            *(Click the name of each method to see more details and examples)*
The :class:`~.QubitDevice` class
```

Reply:
Just a suggestion here

Reply:
This will cause it to render as a block quote, however, which might make it stand out too much